sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkLauncher.java

```diff
@@ -1154,4 +1140,81 @@ public NexmarkPerf run(NexmarkConfiguration runConfiguration) {
       queryName = null;
     }
   }
+
+  private boolean isSql() {
+    return SQL.equalsIgnoreCase(options.getQueryLanguage());
+  }
+
+  private NexmarkQueryModel getNexmarkQueryModel() {
+    List<NexmarkQueryModel> models = createQueryModels();
+    return models.get(configuration.query);
+  }
+
+  private NexmarkQuery getNexmarkQuery() {
+    List<NexmarkQuery> queries = createQueries();
+
+    if (options.getQuery() >= queries.size()) {
+      throw new UnsupportedOperationException(
+          "Query " + options.getQuery()
+          + " is not implemented yet");
+    }
+
+    return queries.get(configuration.query);
+  }
+
+  private List<NexmarkQueryModel> createQueryModels() {
+    return isSql()
+        ? createSqlQueryModels()
+        : createJavaQueryModels();
+  }
+
+  private List<NexmarkQueryModel> createSqlQueryModels() {
+    return Arrays.asList(
+        null, null, null, null, null, null, null, null, null, null, null, null);
+  }
+
+  private List<NexmarkQueryModel> createJavaQueryModels() {
+    return Arrays.asList(
```

Comment:
<!--thread_id:cc_154870916_t; commit:964cc527e15c17e26b9584aaaf993cddc225398b; resolved:1-->
<!--section:context-quote-->
> **kennknowles** wrote:
> Nit: `Immutable.list.of(... elems ...)` is a little more readable than `Arrays.asList`.

<!--section:body-->
I agree but `ImmutableList.of()` doesn't allow `null`s. There is an explicit logic to skip query models if they are `null`, so I will keep it for now. Refactoring that bit is out of scope of this PR.